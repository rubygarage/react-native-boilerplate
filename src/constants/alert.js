// eslint-disable-next-line import/prefer-default-export
export const ALERT_TYPES = {
  error: 'error',
  success: 'success',
  info: 'info',
};
