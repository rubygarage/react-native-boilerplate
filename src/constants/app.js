export const GLOBAL_ERROR = '_error';
export const OFFLINE_FLASH_ID = 'offline';
export const OFFLINE_FLASH_PRIORITY = 0;
export const FLASH_LIFETIME = 3000;

export const PLATFORM_TYPE = {
  android: 'android',
  ios: 'ios',
};
