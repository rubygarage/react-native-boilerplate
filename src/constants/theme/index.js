export default {
  colors: {
    black: '#000000',
    white: '#ffffff',
  },
  fonts: {
  },
  sizes: {
  },
};
