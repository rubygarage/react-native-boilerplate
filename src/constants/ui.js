// eslint-disable-next-line import/prefer-default-export
export const POPOVER_PLACEMENT = {
  bottom: 'bottom',
  auto: 'auto',
};
