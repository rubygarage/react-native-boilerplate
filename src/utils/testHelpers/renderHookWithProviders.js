/* eslint-disable import/no-extraneous-dependencies */
import { renderHook } from '@testing-library/react-hooks';
import { ThemeProvider } from 'styled-components';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import mockedTheme from 'utils/testHelpers/mockedTheme';

const store = configureStore()({});
store.dispatch = jest.fn();

const renderHookWithProviders = (hook) => {
  const wrapper = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider theme={mockedTheme}>
        {children}
      </ThemeProvider>
    </Provider>
  );

  return renderHook(hook, { wrapper });
};

export default renderHookWithProviders;
