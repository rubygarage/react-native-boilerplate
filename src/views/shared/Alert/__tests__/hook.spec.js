import { act } from '@testing-library/react-hooks';

import { ALERT_TYPES } from 'constants/alert';
import renderHookWithProviders from 'utils/testHelpers/renderHookWithProviders';

import useContainer from '../hook';

describe('Alert useContainer hook', () => {
  const { result } = renderHookWithProviders(() => useContainer());

  it('matches snapshot', () => {
    expect(result.current).toMatchSnapshot();
  });

  describe('getStyleProps', () => {
    let style = null;

    it('returns info style props', () => {
      const { getStyleProps, theme } = result.current;

      act(() => {
        style = getStyleProps(ALERT_TYPES.info);
      });

      expect(style).toEqual({
        contentColor: theme.colors.blue500,
        backgroundColor: theme.colors.blue50,
        iconName: 'info',
      });
    });

    it('returns error style props', () => {
      const { getStyleProps, theme } = result.current;

      act(() => {
        style = getStyleProps(ALERT_TYPES.error);
      });

      expect(style).toEqual({
        contentColor: theme.colors.red500,
        backgroundColor: theme.colors.red50,
        iconName: 'close',
      });
    });

    it('returns success style props', () => {
      const { getStyleProps, theme } = result.current;

      act(() => {
        style = getStyleProps(ALERT_TYPES.success);
      });

      expect(style).toEqual({
        contentColor: theme.colors.green500,
        backgroundColor: theme.colors.green50,
        iconName: 'tick',
      });
    });
  });
});
