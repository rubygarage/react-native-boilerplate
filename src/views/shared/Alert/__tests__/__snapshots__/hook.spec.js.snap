// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert useContainer hook matches snapshot 1`] = `
Object {
  "getStyleProps": [Function],
  "theme": Object {
    "colors": null,
    "fonts": null,
    "sizes": null,
    "type": "Mocked theme",
  },
}
`;
